specs/waku/waku-1.md

```diff
@@ -0,0 +1,599 @@
+- a) Accretion (including protocol negotiation) over changing data
+- b) When we want to change things, we give it a new name (for example, a version number).
+
+Examples:
+
+- We enable bridging between `shh/6` and `waku/1` until such a time as when we are ready to gracefully drop support for `shh/6` (1, 2, 3).
+- When we add parameter fields, we (currently) do so by accreting them in a list, so old clients can ignore new fields (dynamic list) and new clients can use new capabilities (1, 3).
+- To better support (2) and (3) in the future, we will likely release a new version that gives better support for open, growable maps (association lists or native map type) (3)
+- When we we want to provide a new set of messages that have different requirements, we do so under a new protocol version and employ protocol versioning. This is a form of accretion at a level above - it ensures a client can support both protocols at once and drop support for legacy versions gracefully. (1,2,3)
+
+### Backwards Compatibility
+
+Waku is a different subprotocol from Whisper so it isn't directly compatible. However, the data format is the same, so compatibility can be achieved by the use of a bridging mode as described below. Any client which does not implement certain packet codes should gracefully ignore the packets with those codes. This will ensure the forward compatibility.
+
+### Waku-Whisper bridging
+
+`waku/1` and `shh/6` are different DevP2P subprotocols, however they share the same data format making their envelopes compatible. This means we can bridge the protocols naively, this works as follows.
+
+**Roles:**
+- Waku client A, only Waku capability
+- Whisper client B, only Whisper capability
+- WakuWhisper bridge C, both Waku and Whisper capability
+
+**Flow:**
+1. A posts message; B posts message.
+2. C picks up message from A and B and relays them both to Waku and Whisper.
+3. A receives message on Waku; B on Whisper.
+
+**Note**: This flow means if another bridge C1 is active, we might get duplicate relaying for a message between C1 and C2. I.e. Whisper(<>Waku<>Whisper)<>Waku, A-C1-C2-B. Theoretically this bridging chain can get as long as TTL permits.
+
+### Forward Compatibility
+
+It is desirable to have a strategy for maintaining forward compatibility between `waku/1` and future version of waku. Here we outline some concerns and strategy for this.
+
+- **Connecting to nodes with multiple versions:** The way this SHOULD be accomplished in the future is by negotiating the versions of subprotocols, within the `hello` message nodes transmit their capabilities along with a version. As suggested in [EIP-8](https://eips.ethereum.org/EIPS/eip-8), if a node connects that has a higher version number for a specific capability, the node with a lower number SHOULD assume backwards compatiblity. The node with the higher version will decide if compatibility can be assured between versions, if this is not the case it MUST disconnect.
```

Comment:
The last part of this sentence is contradicting with how we deal with the version number now, and should be removed or reformulated.